akka-projection-kafka/src/main/scala/akka/projection/kafka/internal/KafkaSourceProviderImpl.scala

```diff
@@ -91,12 +91,11 @@ import org.apache.kafka.common.TopicPartition
       readOffsets()
         .flatMap {
           case Some(groupOffsets) =>
-            Future.successful(groupOffsets.entries.flatMap {
-              case (topicPartitionKey, offset) =>
-                val tp = topicPartitionKey.tp
-                if (assignedTps.contains(tp)) Map(tp -> offset)
-                else Map.empty
-            })
+            val filteredMap = groupOffsets.entries.collect {
+              case (topicPartitionKey, offset) if assignedTps.contains(topicPartitionKey.tp) =>
+                (topicPartitionKey.tp -> offset)
+            }
+            Future.successful(filteredMap)
```

Comment:
Here the inference got broken, something with CanBuild stuff. It turns out that we don't need to `flatMap` and return `Map.empty` for the cases we want to discard. `Map.collect` is all what we need.